src/MacVim/MMWindowController.m

```diff
@@ -1567,6 +1567,12 @@ - (void)updateToolbar
 
 - (BOOL)maximizeWindow:(int)options
 {
+    if (floor(NSAppKitVersionNumber) > NSAppKitVersionNumber10_10_Max) {
```

Comment:
I had exactly the same thought about that. However I couldn't find any situation that the code is needed on El Capitan. If you find anything, please feel free to create issues.